File: atcoder/abc145B_echo.hs
-- unihernandez22
-- https://atcoder.jp/contests/abc145/tasks/abc145_b
-- implementation

main = do
  li <- getLine
  let n = read li
  li <- getLine
  putStrLn $
    if n `mod` 2 == 1 then
      "No"
    else if take (n `div` 2) li ==
            drop (n `div` 2) li then
      "Yes"
    else
      "No"
  

File: atcoder/abc145B_echo.ml
(*(* vicfred
 * https://atcoder.jp/contests/abc145/tasks/abc145_b
 * string manipulation
 * *)
let n = read_int()
let s = read_line()
let main =
    print_endline @@
    if String.(sub s 0 @@ n/2 = sub s (n/2) @@ (n+1)/2) then "Yes"
    else "No"

